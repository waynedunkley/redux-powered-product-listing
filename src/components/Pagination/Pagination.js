import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { Link } from "react-router-dom"
import divStyles from "./Pagination.style"

const StyledDiv = styled.div`
  ${divStyles};
`

const propTypes = {
  current: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
}

const Pagination = ({ current, totalPages }) => {
  let links = []
  if (current === 1) {
    links = [1, 2, 3]
  } else if (current === totalPages) {
    links = [totalPages - 2, totalPages - 1, totalPages]
  } else {
    links = [current - 1, current, current + 1]
  }
  return (
    <StyledDiv>
      <ul>
        <li className={`prev ${current === 1 && "active"}`}>
          <Link to={`/${current - 1}`}>{`< Previous page`}</Link>
        </li>
        {links &&
          links.map(link => (
            <li key={link} className={`num ${current === link && "active"}`}>
              <Link to={`/${link}`}>{link}</Link>
            </li>
          ))}
        <li className={`next ${current === totalPages && "active"}`}>
          <Link to={`/${current + 1}`}>{`Next page >`}</Link>
        </li>
      </ul>
    </StyledDiv>
  )
}

Pagination.propTypes = propTypes

export default Pagination
