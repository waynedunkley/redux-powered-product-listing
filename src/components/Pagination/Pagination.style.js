const divStyles = ({ theme: { colors } }) => `
  display: flex;
  justify-content: flex-end;
  margin-bottom: 50px;
  ul {
    display: flex;
    list-style: none;
    padding: 0;
    margin: 0;
    li {
      border-bottom: 2px solid transparent;
      &.active {
        pointer-events: none;
        &.next,
        &.prev {
          opacity: 0.5;
        }
      }
      a {
        display: block;
        color: ${colors.dark};
        padding: 4px 10px;
      }
      &.num {
        &.active {
          background-color: ${colors.light};
          border-bottom-color: ${colors.dark};
        }
      }
    }
  }
`

export default divStyles
