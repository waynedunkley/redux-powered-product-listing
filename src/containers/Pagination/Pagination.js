import React from "react"
import { Container } from "components"

const Pagination = () => (
  <Container>
    <p>Pagination</p>
  </Container>
)

export default Pagination
