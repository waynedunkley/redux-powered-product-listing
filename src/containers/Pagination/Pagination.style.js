const divStyles = ({ theme: { colors } }) => `
  display: flex;
  justify-content: flex-end;
`

export default divStyles
