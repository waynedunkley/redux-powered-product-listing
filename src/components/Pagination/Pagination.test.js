import React from "react"
import { shallow } from "enzyme"
import Pagination from "./Pagination"

it("Pagination renders correctly", () => {
  const current = 3
  const totalPages = 9
  const wrapper = shallow(
    <Pagination current={current} totalPages={totalPages} />,
  )
  expect(wrapper.find("li")).toHaveLength(5)
  expect(wrapper.find("li.active").contains(current)).toBeTruthy()
  expect(wrapper.find("li.prev").contains("< Previous page")).toBeTruthy()
  expect(wrapper.find("li.next").contains("Next page >")).toBeTruthy()
})
