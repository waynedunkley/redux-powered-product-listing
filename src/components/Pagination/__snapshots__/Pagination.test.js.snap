// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    current={3}
    totalPages={9}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ul>
        <li
                className="prev false"
        >
                <Link
                        replace={false}
                        to="/2"
                >
                        &lt; Previous page
                </Link>
        </li>
        <li
                className="num false"
        >
                <Link
                        replace={false}
                        to="/2"
                >
                        2
                </Link>
        </li>
        <li
                className="num active"
        >
                <Link
                        replace={false}
                        to="/3"
                >
                        3
                </Link>
        </li>
        <li
                className="num false"
        >
                <Link
                        replace={false}
                        to="/4"
                >
                        4
                </Link>
        </li>
        <li
                className="next false"
        >
                <Link
                        replace={false}
                        to="/4"
                >
                        Next page &gt;
                </Link>
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="prev false"
>
            <Link
                        replace={false}
                        to="/2"
            >
                        &lt; Previous page
            </Link>
</li>,
          Array [
            <li
              className="num false"
>
              <Link
                            replace={false}
                            to="/2"
              >
                            2
              </Link>
</li>,
            <li
              className="num active"
>
              <Link
                            replace={false}
                            to="/3"
              >
                            3
              </Link>
</li>,
            <li
              className="num false"
>
              <Link
                            replace={false}
                            to="/4"
              >
                            4
              </Link>
</li>,
          ],
          <li
            className="next false"
>
            <Link
                        replace={false}
                        to="/4"
            >
                        Next page &gt;
            </Link>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/2"
>
              &lt; Previous page
</Link>,
            "className": "prev false",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "< Previous page",
              "replace": false,
              "to": "/2",
            },
            "ref": null,
            "rendered": "< Previous page",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/2"
>
              2
</Link>,
            "className": "num false",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "replace": false,
              "to": "/2",
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/3"
>
              3
</Link>,
            "className": "num active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 3,
              "replace": false,
              "to": "/3",
            },
            "ref": null,
            "rendered": 3,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/4"
>
              4
</Link>,
            "className": "num false",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 4,
              "replace": false,
              "to": "/4",
            },
            "ref": null,
            "rendered": 4,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/4"
>
              Next page &gt;
</Link>,
            "className": "next false",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Next page >",
              "replace": false,
              "to": "/4",
            },
            "ref": null,
            "rendered": "Next page >",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ul>
          <li
                    className="prev false"
          >
                    <Link
                              replace={false}
                              to="/2"
                    >
                              &lt; Previous page
                    </Link>
          </li>
          <li
                    className="num false"
          >
                    <Link
                              replace={false}
                              to="/2"
                    >
                              2
                    </Link>
          </li>
          <li
                    className="num active"
          >
                    <Link
                              replace={false}
                              to="/3"
                    >
                              3
                    </Link>
          </li>
          <li
                    className="num false"
          >
                    <Link
                              replace={false}
                              to="/4"
                    >
                              4
                    </Link>
          </li>
          <li
                    className="next false"
          >
                    <Link
                              replace={false}
                              to="/4"
                    >
                              Next page &gt;
                    </Link>
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="prev false"
>
              <Link
                            replace={false}
                            to="/2"
              >
                            &lt; Previous page
              </Link>
</li>,
            Array [
              <li
                className="num false"
>
                <Link
                                replace={false}
                                to="/2"
                >
                                2
                </Link>
</li>,
              <li
                className="num active"
>
                <Link
                                replace={false}
                                to="/3"
                >
                                3
                </Link>
</li>,
              <li
                className="num false"
>
                <Link
                                replace={false}
                                to="/4"
                >
                                4
                </Link>
</li>,
            ],
            <li
              className="next false"
>
              <Link
                            replace={false}
                            to="/4"
              >
                            Next page &gt;
              </Link>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/2"
>
                &lt; Previous page
</Link>,
              "className": "prev false",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "< Previous page",
                "replace": false,
                "to": "/2",
              },
              "ref": null,
              "rendered": "< Previous page",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/2"
>
                2
</Link>,
              "className": "num false",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 2,
                "replace": false,
                "to": "/2",
              },
              "ref": null,
              "rendered": 2,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/3"
>
                3
</Link>,
              "className": "num active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 3,
                "replace": false,
                "to": "/3",
              },
              "ref": null,
              "rendered": 3,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/4"
>
                4
</Link>,
              "className": "num false",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 4,
                "replace": false,
                "to": "/4",
              },
              "ref": null,
              "rendered": 4,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/4"
>
                Next page &gt;
</Link>,
              "className": "next false",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Next page >",
                "replace": false,
                "to": "/4",
              },
              "ref": null,
              "rendered": "Next page >",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
